app/Commands/Composer.php

```diff
@@ -30,8 +28,10 @@ class Composer extends Command
      *
      * @return mixed
      */
-    public function handle(CommandExecutor $executor)
+    public function handle()
     {
-        return $executor->run(new ComposerCommand($this->getArgs()));
+        return $this->commandExecutor->run(
+            ComposerBuilder::make($this->getArgs())
```

Comment:
Since the handle method is similar in almost all commands, can't we pass the responsibility of if to the parent class? This way, in the child class we would just return wich builder to use. Just an idea.

Reply:
This is a nice thought, a lot of the commands will have a single Builder that we want to execute, so I guess we can simplify all of them by having a new Command base class like SingleBuilderCommand, that implements this handle already, but asks for a getBuilderClass()... saves code, but not sure if the trade off pays out in terms of clarity.

Reply:
We can also add a method on the base class that runs the command, passing the builder as a parameter. It saves code, keeps the clarity and maybe it can create a better place to inject the command executor, solving the comment above about the file 'app/Commands/Command.php'. Also, it will be possible to protect $commandExecutor, but I don't know if it's really necessary to protect it...